### ISSUE TITLE
[WebExt] User Scripts docs tasks, Fx 70 (high priority)

### ISSUE THREAD

**Issue by chrisdavidmills** [mdn/sprints#2136]
Trello: https://trello.com/c/vsKZdybA/115-page-actions-and-userscripts-fx-70

# Acceptance criteria

1. - [x] [Enable userScripts API on release channels](Enable userScripts API on release channels) (https://bugzilla.mozilla.org/show_bug.cgi?id=1437098) (see [comment 26](https://bugzilla.mozilla.org/show_bug.cgi?id=1437098#c26) - there is an existing doc, but it's not good enough - needs to be expanded upon)
2. - [ ]  [[meta] userScripts API enhancements and follow-ups](https://bugzilla.mozilla.org/show_bug.cgi?id=1595853)
2. - [x] [[META] Implement a userScripts API](https://bugzilla.mozilla.org/show_bug.cgi?id=1437098) (the same docs need, but needs DDC'ing once work is done).

## Notes

https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/userScripts definitely needs improvement:

* missing a page for `onBeforeScript`
* Many of the descriptions seem to be about contentScripts, on both this page and the subpages.
* The page needs an example, shown here and added to the webextensions examples repo. The subpages also need examples.
* BCD is missing from all pages
* You could probably do with writing a tutorial to show how to use userScripts, in a similar style to https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Content_scripts

# Acceptance criteria

- [x] BCD PR created, reviewed and merged
- [x] onBeforeScript page created
- [x] All pages reviewed and edited to remove any inappropriate reference to contentScripts
- [x] Tutorial created and reviewed
- [x] Example extension created and added to web extension example repo
- [ ] appropriate additions to Web Extensions sidebar made and merged

**Comment by irenesmith** [comment]
BCD [PR #4768](https://github.com/mdn/browser-compat-data/pull/4768/files) Approved and merged.

**Comment by irenesmith** [comment]
Resolving missing sidebar entries for userScripts methods, events, and types

**Comment by irenesmith** [comment]
Tutorial created but not reviewed. The single task that has not been done at all is the PR for adding the new example.

**Comment by irenesmith** [comment]
Created this PR for the new example: [PR 426](https://github.com/mdn/webextensions-examples/pull/426)

**Comment by irenesmith** [comment]
My tutorial page didn't update properly. The page [Working with userScripts](https://wiki.developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/userScripts/Working_with_userScripts) not only is only a skeleton but there is no history on the finished article. (How that can happen, I don't understand.)

Recreating...

**Comment by chrisdavidmills** [comment]
@irenesmith are you going to finish off this tutorial any time soon?

**Comment by jmswisher** [comment]
@irenesmith What still needs to be done on this to remove the Draft banner?

**Comment by jmswisher** [comment]
Irene will create a new issue for replacing the example. Otherwise, this is done.

**Comment by rebloor** [comment]
MDN content has been updated, waiting on final review/confirmation from Luca.  New copy of the example has been submitted with feedback about the readme content addressed, with Will for review.
